common/src/main/java/com/microsoft/identity/common/internal/ui/browser/CustomTabsManager.java

```diff
@@ -160,8 +174,13 @@ public CustomTabsClient getClient() {
      * Method to unbind custom tabs service {@link androidx.browser.customtabs.CustomTabsService}.
      */
     public synchronized void unbind() {
-        if (mContextRef.get() != null && mCustomTabsServiceIsBound) {
-            mContextRef.get().unbindService(mCustomTabsServiceConnection);
+        final Context context = mContextRef.get();
+        if (context != null && mCustomTabsServiceIsBound) {
+            try {
+                context.unbindService(mCustomTabsServiceConnection);
```

Comment:
The documentation doesn't indicate that "unbindService" throws an exception.

Reply:
The documentation might not indicate it.  Two things:

1) Runtime exceptions aren't checked exceptions, and can be thrown from anywhere without appearing in the method signature.  IllegalArgumentException is a runtime exception.

2) This is clearly occurring:

8-06 18:39:00.814 10299 27760 27760 E AndroidRuntime: 	at android.app.LoadedApk.forgetServiceDispatcher(LoadedApk.java:1901)
08-06 18:39:00.814 10299 27760 27760 E AndroidRuntime: 	at android.app.ContextImpl.unbindService(ContextImpl.java:1954)
08-06 18:39:00.814 10299 27760 27760 E AndroidRuntime: 	at android.content.ContextWrapper.unbindService(ContextWrapper.java:810)
08-06 18:39:00.814 10299 27760 27760 E AndroidRuntime: 	at com.microsoft.identity.common.internal.ui.browser.CustomTabsManager.unbind(CustomTabsManager.java:2)
08-06 18:39:00.814 10299 27760 27760 E AndroidRuntime: 	at com.microsoft.identity.common.internal.ui.browser.BrowserAuthorizationStrategy.dispose(BrowserAuthorizationStrategy.java:3)

3) In the context implementation this throws here:

https://cs.android.com/android/platform/superproject/+/master:frameworks/base/core/java/android/app/LoadedApk.java;drc=875888eff38e3a6c1369bbb9c2336e7b97758c51;l=1793

Given this, do you have a suggestion?

Reply:
Thanks for writeup and offline discussion.  Approving.  Thanks

Reply:
> Feels like something strange is happening here inter of service connection lifecycle. It's like the connection is being closed before it's actually bound/connected.... i found similar exceptions when things happen out of order.



> Feels like something strange is happening here inter of service connection lifecycle. It's like the connection is being closed before it's actually bound/connected.... i found similar exceptions when things happen out of order.

I agree.  There's something fishy here.  Ideas for either logging to catch the issue or methods to reproduce are welcome.